algo/src/main/java/com/hartwig/actin/algo/evaluation/complication/PatternMatcher.java

```diff
@@ -30,4 +35,37 @@ public static boolean isMatch(@NotNull String term, @NotNull Set<List<String>> p
 
         return false;
     }
+
+    public static Set<String> findComplicationNamesMatchingCategories(PatientRecord record,
+                                                                      List<String> categorySearchTerms) {
+        return findComplicationsMatchingCategories(record, categorySearchTerms).orElse(Stream.empty())
+                .map(Complication::name)
+                .collect(Collectors.toSet());
+    }
+
+    public static Set<String> findComplicationCategoriesMatchingCategories(PatientRecord record,
+                                                                            List<String> categorySearchTerms) {
+        return findComplicationsMatchingCategories(record, categorySearchTerms).orElse(Stream.empty())
+                .flatMap(complication -> complication.categories().stream())
+                .collect(Collectors.toSet());
+    }
+
+    private static Optional<Stream<ImmutableComplication>> findComplicationsMatchingCategories(PatientRecord record,
```

Comment:
Naming could be improved, similar to above comment

(side note: really nice to see such examples of streams!)